test/testcafe/spec/StateTokenRefreshUseCases_spec.js

```diff
@@ -0,0 +1,75 @@
+import { RequestMock, RequestLogger } from 'testcafe';
+import SelectAuthenticatorPageObject from '../framework/page-objects/SelectAuthenticatorPageObject';
+import IdentityPageObject from '../framework/page-objects/IdentityPageObject';
+import xhrIdentify from '../../../playground/mocks/data/idp/idx/identify';
+import xhrErrorIdentify from '../../../playground/mocks/data/idp/idx/error-identify-access-denied';
+import xhrAuthenticatorVerifySelect from '../../../playground/mocks/data/idp/idx/authenticator-verification-select-authenticator';
+
+const identifyMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentify)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrErrorIdentify, 403);
+
+const identifyThenSelectAuthenticatorMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(xhrIdentify)
+  .onRequestTo('http://localhost:3000/idp/idx/identify')
+  .respond(xhrAuthenticatorVerifySelect);
+
+const identifyRequestLogger = RequestLogger(
+  /idx\/identify/,
+  {
+    logRequestBody: true,
+    stringifyRequestBody: true,
+  }
+);
+
+fixture('Statetoken Page Refresh');
+
+async function setup(t) {
+  const identityPage = new IdentityPageObject(t);
+  await identityPage.navigateToPage();
+  return identityPage;
+}
+
+async function setupA(t) {
+  const identityPage = new SelectAuthenticatorPageObject(t);
+  await identityPage.navigateToPage();
+  return identityPage;
+}
+
+test.requestHooks(identifyRequestLogger, identifyMock)('should set and remove statetoken in sessionStorage', async t => {
+  const identityPage = await setup(t);
+  let stateToken = await t.eval(() => sessionStorage.getItem('okta-siw-state-token'));
+  await t.expect(stateToken !== null).eql(true);
+  await identityPage.fillIdentifierField('Test Identifier');
+  await identityPage.clickNextButton();
+  // On terminal error state token should be cleared
```

Comment:
It's unclear to me that we need to clear session storage at this point cause there are many reasons that identify step got error. (one example I would guess is user put wrong username and backend responds error, but user can correct that with right user name)

Reply:
yeah make sense.. changed it to success case instead of user error.